File: solutions/ocaml/nth-prime/1/nth_prime.ml
let has_divisor_in_lst n lst = 
  List.mem true (List.map (fun k -> n mod k = 0) lst)

let nth_prime num = 
  if num<=0 then 
    Error("there is no zeroth prime")
  else
    let rec nth cand primes = 
      if List.length primes >= num then (cand - 2) 
      else if has_divisor_in_lst cand primes then nth (cand + 2) primes
      else nth (cand + 2) (cand :: primes)
    in    
    if num=1 then Ok 2
    else if num=2 then Ok 3
    else
      Ok (nth 5 [2 ; 3])
    

File: solutions/dart/nth-prime/1/lib/nth_prime.dart
class NthPrime {
  
  int prime(int N){
    if(N<1)
      throw ArgumentError("There is no zeroth prime");
    if(N<3)
      return N==1 ? 2 : 3; 
    
    var primes = [2,3];
    int count = 2;
    for(var cand=5; count<=N; cand += 2){
      var isPrime = true;
      for(var p in primes){
        if(0==cand % p){
          isPrime = false;
          break;
        }        
      }
      if(isPrime){
        primes.add(cand);
        count ++;
      }
    }
    return primes[N-1];
  }
  
}
